examples/testing_ioc/pva_testing_ioc.py

```diff
@@ -0,0 +1,78 @@
+from p4p.client.thread import Context
+from p4p.nt import NTNDArray, NTScalar
+from p4p.server import Server, ServerOperation
+from p4p.server.thread import SharedPV
+import numpy as np
+import random
+import threading
+
+
+class Handler(object):
+    """ A handler for dealing with put requests to our test PVs """
+
+    def put(self, pv: SharedPV, op: ServerOperation) -> None:
+        """ Called each time a client issues a put operation on the channel using this handler """
+        pv.post(op.value())
+        op.done()
+
+
+class PVServer(object):
+    """ A simple server created by p4p for making some PVs to read and write test data to """
+    def __init__(self):
+        self.create_pvs()
+        self.context = Context('pva', nt=False)  # Disable automatic value unwrapping
+        self.event = threading.Event()
+        print('Creating testing server...')
+        self.server_thread = threading.Thread(target=self.run_server)
+        self.server_thread.start()
+        self.update_thread = threading.Thread(target=self.update_pvs, daemon=True)
+        self.update_thread.start()
+
+    def create_pvs(self) -> None:
+        """ Create a few PVs for interacting with """
+        handler = Handler()  # A simple default handler that will just post PV updates is sufficient here
+        # A scalar int value with some alarm limits set
+        nt = NTScalar("i", display=True, control=True, valueAlarm=True)
+        initial = nt.wrap({'value': 5, 'valueAlarm': {'lowAlarmLimit': 2, 'lowWarningLimit': 3,
+                                                      'highAlarmLimit': 9, 'highWarningLimit': 8}})
+
+        self.int_value = SharedPV(handler=handler,
```

Comment:
Seems like it'd be a good idea to create all all NT data type variants (float, string, arrays of each, ...) and add them to the test display for completeness

Reply:
Added additional types here: d2b67ce3c48ef2951a193dff8117829136aee435
Added additional displays here: 15dabd011222d1c88a3474e8c3c28ffa8ae11157 and here: a451473b52362309e8cf795e0e573809f46c429e